### DISCUSSION TITLE
location info in onnx

### DISCUSSION THREAD

**Discussion by caoimhinuibrian** [onnx/onnx#2993] [Ideas]
# Feature Request

### System information
Any

### What is the problem that this feature solves?
Currently there appears to be no way to express location information along with an ONNX operator. Frameworks that export
ONNX representations may want to do this for later debug support or for error reporting in a backend.

### Describe the alternatives you have considered
Protobuf supports a class DescriptorProtos.SourceCodeInfo.Location which possibly could be used, but really I'm looking to start discussion.

### Describe the feature
Improvement to usability.

### Will this influence the current api?
Yes. Probably will add optional parameters to some API calls.

### Feature Area
backend, best practices, converters

### Are you willing to contribute it (Y/N):
Perhaps

**Comment by prasanthpul** [comment]
can you elaborate on what you mean by location information?

**Comment by caoimhinuibrian** [comment]
Well, any data that expresses information on where in a source file the operator originated. Depending on what exporter produced the ONNX protobuf that could vary, but in a simple case it could be a python file name and a line number and possibly offset. One can imagine more complex scenarios if the protobuf is generated automatically from some higher level tool. The location can be generalized so that different producers are allowed to use formats that makes sense to their users. From an ONNX point of view, all that is needed is a way of storing it in the protobuf so that downstream tools that are interested can work with it. For example, onnx-mlir could encode this as a location in MLIR and it would be passed through to LLVM.

**Comment by askhade** [comment]
Converting this issue to discussion

**Comment by faxu** [comment]
ONNX supports custom metadata...can this information just be added there?

**Comment by askhade** [comment]
Can you elaborate how this location info can be useful?
